index/miner/meta.go

```diff
@@ -50,6 +49,7 @@ func (mi *MinerIndex) metaWorker() {
 			newIndex, err := updateMetaIndex(mi.ctx, mi.api, mi.h, mi.lr, addrs)
 			if err != nil {
 				log.Errorf("error when updating meta index: %s", err)
+				break
```

Comment:
If for some reason generating a new index fails, log the error and break here. If we continue we would override the current-good-state-index with an empty one and lost things. Better to keep it outdated than to lose it.